apps/server/src/classes/data-provider/__tests__/DataProvider.utils.test.ts

```diff
@@ -210,3 +220,18 @@ describe('safeMerge', () => {
     expect(result.customFields).toEqual(expected);
   });
 });
+
+describe('drop keys', () => {
+  let fullEvent: OntimeEvent = { id: '1', cue: '#1', type: SupportedEvent.Event, ...event };
+  let dbEvent = fullEvent as DatabaseOntimeEvent;
+  beforeEach(() => {
+    fullEvent = { id: '1', cue: '#1', type: SupportedEvent.Event, ...event };
+    dbEvent = fullEvent;
+    delete (dbEvent as OntimeEvent).delay;
+  });
+  it('drops delay key', () => {
+    const testRundown: OntimeRundown = [{ ...fullEvent }];
+    const result = rundownToDatabaseRundown(testRundown);
+    expect(result).toEqual([{ id: '1', ...dbEvent }]);
+  });
+});
```

Comment:
suggestion: this test seems a bit complicated for what it achieves
suggestions:
- we can work on making things reusable later
- remember to name the tests in a way that they tell us what is under test

maybe something like

```suggestion
describe('rundownToDatabaseRundown()', () => {
  it('converts the rundown to DAO type by removing the delay', () => {
    const testEvent: OntimeEvent = { ...event, id: '1', cue: '#1' };

    const testRundown: OntimeRundown = [testEvent];
    const result = rundownToDatabaseRundown(testRundown);
    // remove the delay from the event
    const { delay, ...matchObject } = testEvent;
    expect(result).toMatchObject([matchObject]);
  });
});
```

Reply:
Yeah I thought I was going to make more tests and would need fullEvent multiple times, but can't come up with any more relevant test, can you?

Reply:
you could assert that the objects are not the same (ie, the function does not mutate the original object
but I am not sure this is necessary. it is probably fine as it is